src/plugins/sailfishoswizards/include/models/externallibrary.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Qt Creator plugin with wizards of the Sailfish OS projects.
+ * Copyright © 2020 FRUCT LLC.
+ */
+
+#ifndef EXTERNALLIBRARY_H
+#define EXTERNALLIBRARY_H
+
+#include <QStringList>
+
+namespace SailfishOSWizards {
+namespace Internal {
+
+/*!
+ * \brief ExternalLibrary class is a model for external library.
+ * The class contains an information about the Qt external library.
+ */
+class ExternalLibrary
+{
+
+public:
+    ExternalLibrary() {}
+
+    void setName(const QString &name);
+    void setLayoutName(const QString &layoutName);
+    void setQtList(QStringList qtList);
+    void setDependencyList(QStringList dependencyList);
+    void setTypesList(QStringList types);
+    void setYamlHeader(const QString &yamlHeader);
+    void setSpecHeader(const QString &specHeader);
+
+    QString getName() const;
+    QString getLayoutName() const;
+    QStringList getQtList() const;
+    QStringList getDependencyList() const;
+    QStringList getTypesList() const;
+    QString getYamlHeader() const;
+    QString getSpecHeader() const;
```

Comment:
The part of updating .spec/.yaml seems to not be implemented yet. This would be also the most challenging part to abstract from to allow upstreaming. Ideas?

Reply:
Methods for spec- and yaml-files are used inside other wizards for creating Yaml and Spec files. Here I forgot to remove them from the file. Do I should to remove them from this pull request?

Reply:
Yes, if the code is only needed for future PRs then I would prefer only including it in the future PRs, i.e., removing it from this PR.

But my understanding was the .spec/.yaml would be updated when adding a new QML file that brings new dependencies. Hence my question if you have any idea how to abstract that part for upstreaming. Note on that, that it should be enough to only care about YAML and leave .spec untouched, as that one would be updated from the YAML on next build anyway.

Reply:
Qt Creator can be used also on projects which only use .spec (and no .yaml), so it's not good to leave .spec untouched. Maybe there could be some logic, that if the project has .yaml file, then .spec could be left alone. But then again, that would just add unnecessary complications to the code, so I wouldn't do even that.

Reply:
Leaving .spec untouched is perfectly OK, because the yaml-spec cooperation it is designed for that. The .spec is not completely overwritten based on the yaml. The yaml packaging format cover only subset of .spec. In advanced cases, the generated .spec is to be hand-edited and `specify` will preserve those modifications next time it updates it from yaml.

YAML packaging format exists exactly for the advantage of being more easily editable programmatically compared to .spec. It has been always the plan to develop wizards and similar tools that would manipulate the YAML only.

Reply:
@martyone @vigejolla What if we will add dependencies to the YAML and/or SPEC files if a project is opened in Sailfish Qt Creator? But if a project is not opened this wizard will only create a new QML-file.

Reply:
Something like that makes sense, but it shouldn't be controlled by the fact that a project is open. There should be the option to (not) register the new file to a project and/or VCS on the last page. At least when I checked with some existing wizard there is such an option.

Reply:
@martyone Can you tell how we can write a code for editing the YAML (and SPEC) file after going through the whole wizard? 

We think that we should add dependencies when the user clicks "Ok" on the "Summary" page. Class `JsonSummaryPage` contains inside the `projectexplorer` plugin and we think that our code is too specific to add this into the standard `JsonSummaryPage`. Maybe we should to inherit the `JsonSummaryPage` class, implement the YAML file editing and use our class in the wizard? Or maybe do you have some another idea?

Reply:
Nope, I don't have a better idea. Subclassing `JsonSummaryPage` seems to be the only option. (I would expect some generic mechanism being available for that, similar to that one for generating files, but I guess they had no use case for that and so the registration to project is done from `JsonSummaryPage`.)

Reply:
Don't forget just YAML needs to be updated. Spec will receive the modifications automatically.

Reply:
Another option is to create the whole wizard with all pages manually inside the `sailfishoswizards` plugin without using the `wizard.json` file. What about this idea?

> Don't forget just YAML needs to be updated. Spec will receive the modifications automatically.

But what about projects that use only SPEC files? Maybe they cannot be created by default, but users can remove YAML manually and use only SPEC.

Reply:
We should definitely do json based wizards - upstream moved from C++ based wizards long time ago.

As I wrote earlier, one reason for supporting YAML packaging is to avoid the need to edit .spec programmatically. In that sense we should only care about packages that use YAML.

Reply:
> We should definitely do json based wizards - upstream moved from C++ based wizards long time ago.

So, in this case we need to add `PROJECTEXPLORER_EXPORT` macros to the `JsonSummaryPage` class declaration to have a possibility to inherit this class inside `sailfishoswizards` plugin. Can we make this change inside upstream?

Reply:
The best is to ask upstream. Possibly getting some better suggestions from them :)

Reply:
We added PR about adding the `PROJECTEXPLORER_EXPORT` macros to the `JsonSummaryPage` class: https://github.com/sailfishos/sailfish-qtcreator/pull/265